WordPress/Classes/ViewRelated/Media/MediaLibraryViewController.swift

```diff
@@ -415,9 +415,12 @@ class MediaLibraryViewController: WPMediaPickerViewController {
     private func registerUploadCoordinatorObserver() {
         uploadObserverUUID = MediaCoordinator.shared.addObserver({ [weak self] (media, state) in
             switch state {
-            case .progress(let progress) :
-                self?.updateCellProgress(progress, for: media)
-                break
+            case .progress(let progress):
+                if media.remoteStatus == .failed {
```

Comment:
But we shouldn't get a progress callback after a media has failed?

Reply:
I know but we are receiving it, there must be some race condition somewhere between progress report and failure report. I tried several options to avoid the racing condition but i think this is the best safeguard.